components/WorkoutForm.js

```diff
@@ -0,0 +1,268 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { exercises } from "@/lib/exercises";
+import findExerciseById from "@/utils/findExerciseById";
+
+// Styled Components
+const Header = styled.header`
+  text-align: center;
+  margin-bottom: 1rem;
+`;
+
+const WrapperFilter = styled.div`
+  width: 100%;
+  height: 100%;
+  position: fixed;
+  top: 0;
+  z-index: 1;
+  background: rgba(0, 0, 0, 0.6);
+  padding: 1rem 3rem;
+`;
+
+const WrapperForm = styled.div`
+  display: flex;
+  justify-content: center;
+  background-color: white;
+  border-radius: 10px;
+  padding: 1rem;
+`;
+
+const FormContainer = styled.div`
+  width: 100%;
+  max-width: 800px;
+  padding: 1rem;
+  margin: 0 auto;
+`;
+
+const Fieldset = styled.fieldset`
+  border: none;
+  margin-bottom: 1rem;
+  padding: 1rem;
+  border-radius: 8px;
+  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
+`;
+
+const Legend = styled.legend`
+  font-weight: bold;
+  margin-bottom: 0.5rem;
+`;
+
+const Label = styled.label`
+  display: block;
+  margin-bottom: 1rem;
+  font-weight: bold;
+`;
+
+const Input = styled.input`
+  width: 100%;
+  padding: 0.5rem;
+  margin-top: 0.5rem;
+  border-radius: 4px;
+  border: 1px solid #ddd;
+`;
+
+const Select = styled.select`
+  width: 100%;
+  padding: 0.5rem;
+  margin-top: 0.5rem;
+  border-radius: 4px;
+  border: 1px solid #ddd;
+`;
+
+const Button = styled.button`
```

Comment:
Again, not necessary, but maybe you could be a bit more precise with the naming of the button? 😁 Something like AddButton?